spec/front_desk_spec.rb

```diff
@@ -0,0 +1,398 @@
+        @concierge.add_reservation(reservation_5)
+        @concierge.add_block_reservations
+
+        all_block_rooms = @concierge.all_rooms_in_blocks(Date.new(2019, 7, 6))
+        expect(all_block_rooms).must_be_kind_of Array
+        expect(all_block_rooms.length).must_equal 2
+        expect(all_block_rooms[0]).must_equal 17
+        expect(all_block_rooms[1]).must_equal 18
+      end
+    end
+
+    describe "I cannot create another hotel block for unavailable room for that specific date" do
+      before do
+        @concierge = build_front_desk
+      end
+      it "will raise an error if I try to create another hotel block for unavailable room for that specific date" do
+        reservation_32 = @concierge.make_reservation(
+          3065,
+          5,
+          11,
+          Date.new(2019, 7, 5),
+          Date.new(2019, 7, 8),
+        )
+        @concierge.add_reservation(reservation_32)
+        @concierge.add_block_reservations
+
+        expect {
+          @concierge.make_reservation(
+            3069,
+            2,
+            14,
+            Date.new(2019, 7, 5),
+            Date.new(2019, 7, 8),
+          )
+        }.must_raise ArgumentError
+
+        expect {
+          @concierge.make_reservation(
+            3069,
+            2,
+            12,
+            Date.new(2019, 7, 5),
+            Date.new(2019, 7, 8),
+          )
+        }.must_raise ArgumentError
+
+        expect {
+          @concierge.make_reservation(
+            3069,
+            4,
+            8,
+            Date.new(2019, 7, 5),
+            Date.new(2019, 7, 8),
+          )
+        }.must_raise ArgumentError
+      end
+    end
+
+    describe "make_reservation_in_block method" do
+      before do
+        @concierge = build_front_desk
+      end
+
+      it "will describe reserved rooms in block" do
+        reservation_6 = @concierge.make_reservation(
+          3002,
+          2,
+          17,
+          Date.new(2019, 7, 5),
+          Date.new(2019, 7, 7),
+        )
+        @concierge.add_reservation(reservation_6)
+        @concierge.add_block_reservations
+
+        new_reservation = @concierge.make_reservation_in_block(
+          3003,
+          17,
+          Date.new(2019, 7, 5),
+          Date.new(2019, 7, 7),
+        )
+
+        reserved_room = @concierge.add_block_room_reservation(new_reservation)
+
+        avail_rooms = @concierge.block_room_availability(Date.new(2019, 7, 5))
+
+        expect(@concierge.reserved_rooms_in_blocks).must_be_kind_of Array
+        expect(@concierge.reserved_rooms_in_blocks.length).must_equal 1
+        expect(new_reservation.block_room_number).must_equal 17
+        expect(avail_rooms).must_be_kind_of Array
+        expect(avail_rooms.length).must_equal 1
+        expect {
+          @concierge.make_reservation_in_block(
+            30090,
+            18,
+            Date.new(2019, 7, 5),
+            Date.new(2019, 7, 6),
+          )
+        }.must_raise ArgumentError
```

Comment:
Why should this raise an error? If this is a different case, then you should write a different test for it